Source/Core/Common/BitSet.h

```diff
@@ -98,15 +98,16 @@ class BitSet
   constexpr BitSet(std::initializer_list<int> init)
   {
     for (int bit : init)
-      m_val |= (IntTy)1 << bit;
+      m_val |= static_cast<IntTy>(1) << bit;
```

Comment:
You can directly initialize around the literal with brace initialization:

```suggestion
      m_val |= IntTy{1} << bit;
```

which is effectively the same thing, but shorter and generates a compiler error if the value cant be represented in the casted-to type.

Reply:
Is it always better to do it this way assuming I don't want truncation, or are there any cases where it's better to stick with static_cast?

Reply:
Generally you want to do it this way with constants (*especially in generic code*, the error is usually extremely useful there for identifying oversights) and in constructor initializer lists, i.e.

```cpp
class Thing {
public:
  // While this *is* sketchy, normally it'd be followed with a comment that explains *why*
  // this behavior is desired. This also makes the compiler make the cast explicit as opposed to
  // silently truncating like what would happen if you did 'm_data(value)`,
  // which is useful for identifying meaningful truncation explicitly.
  explicit Thing(uint32_t value) : m_data{static_cast<uint16_t>(value)} {}

private:
  uint16_t m_data{};
}